test/paper_toolbar_test.dart

```diff
@@ -0,0 +1,114 @@
+    });
+
+    test('has expected medium-tall height', () {
+      int old = toolbar.offsetHeight;
+      toolbar.classes.add('medium-tall');
+      expect(toolbar.offsetHeight, equals(old * 2));
+    });
+
+    test('has expected tall height', () {
+      int old = toolbar.offsetHeight;
+      toolbar.classes.add('tall');
+      expect(toolbar.offsetHeight, equals(old * 3));
+    });
+
+    test('distributes nodes to topBar by default', () {
+      DivElement item = new DivElement();
+      Polymer.dom(toolbar).append(item);
+      PolymerDom.flush();
+
+      HtmlElement insertionPoint =
+          Polymer.dom(item).getDestinationInsertionPoints()[0];
+      expect(
+          Polymer.dom(insertionPoint).parentNode, equals(toolbar.$['topBar']));
+    });
+
+    test('distributes nodes with "middle" class to middleBar', () {
+      DivElement item = new DivElement();
+      item.classes.add('middle');
+      Polymer.dom(toolbar).append(item);
+      PolymerDom.flush();
+
+      HtmlElement insertionPoint =
+          Polymer.dom(item).getDestinationInsertionPoints()[0];
+      expect(Polymer.dom(insertionPoint).parentNode,
+          equals(toolbar.$['middleBar']));
+    });
+
+    test('distributes nodes with "bottom" class to bottombar', () {
+      DivElement item = new DivElement();
+      item.classes.add('bottom');
+      Polymer.dom(toolbar).append(item);
+      PolymerDom.flush();
+
+      HtmlElement insertionPoint =
+          Polymer.dom(item).getDestinationInsertionPoints()[0];
+      expect(Polymer.dom(insertionPoint).parentNode,
+          equals(toolbar.$['bottomBar']));
+    });
+  });
+
+  group('a11y', () {
+    test('has role="toolbar"', () {
+      PaperToolbar toolbar = fixture('basic');
+      expect(toolbar.getAttribute('role'), equals('toolbar'),
+          reason: 'should have role="toolbar"');
+    });
+
+    test('children with .title becomes the label', () {
+      PaperToolbar toolbar = fixture('title');
+      expect(toolbar.getAttribute('aria-labelledby'), isNotNull,
+          reason: 'should have aria-labelledby');
+      expect(toolbar.getAttribute('aria-labelledby'),
+          equals(toolbar.querySelector('.title').id),
+          reason: 'aria-labelledby should have the id of the .title element');
+    });
+
+    test('existing ids on titles are preserved', () {
+      PaperToolbar toolbar = fixture('title-with-id');
+      expect(toolbar.getAttribute('aria-labelledby'), isNotNull,
+          reason: 'should have aria-labelledby');
+      expect(toolbar.querySelector('.title').id, equals('title'),
+          reason: 'id should be preserved');
+    });
+
+    test('multiple children with .title becomes the label', () {
+      PaperToolbar toolbar = fixture('multiple-titles');
+
+      expect(toolbar.getAttribute('aria-labelledby'), isNotNull,
+          reason: 'should have aria-labelledby');
+
+      List ids = [];
+      List titles = toolbar.querySelectorAll('.title');
+      titles.forEach((title) {
```

Comment:
Personally I would use `map` to build the ids String, but what you have is fine:

```dart
String ids = titles.map((title) => title.id).join(' ');
```